### ISSUE TITLE
Module Request: AJA FSHDR

### ISSUE THREAD

**Issue by filmwork68** [bitfocus/companion-module-aja-fs#1]
**Describe the feature**
Support for control of the AJA FSHDR for SDR/HDR conversions and control

**Is this platform dependent (windows, mac, ..)?**
No its not, controlled via web GUI

**If documentation is required to implement, do you know where to find it?**
If applicable, add screenshots, protocol description PDFs, etc to help the developers along

**Usecases**
Use this in broadcast trucks and film sets in order to do live color correction on multiple cameras as well as LOG to SDR and HDR Conversions of various types.

**Comment by josephdadams** [comment]
@filmwork68 if you'd still like to see this module created, can you provide a link to the documentation to control it? Without a device to develop/test with, we won't be able to create this unless we have documentation on the command structure. Also, what actions are you looking for (what would you like to be able to control on the unit)?

**Comment by filmwork68** [comment]
@josephdadams They don't have official documentation but I have provided an official list of all the RESTAPI calls commands that the FSHDR Supports from AJA (which you can get to on the FSHDR) itself if you have one. The actions I'm looking to support are changing the input/output color spaces, the color control modes, SDI input/output settings. Basically anything in the restAPI but I know that could be a lot to `implement.

[FSHDR Descriptors.zip](https://github.com/bitfocus/companion-module-requests/files/3800766/FSHDR.Descriptors.zip)

**Comment by josephdadams** [comment]
@filmwork68 thanks! I will take a look at this. It looks similar to how AJA does their Ki Pro's, which I am very familiar with.

**Comment by josephdadams** [comment]
@filmwork68 is there an HTML version of this page you could send me? The XML is a little hard to read. The descriptors page on my Ki Pro is very easy to read in a browser, so I was just wondering. Thanks!

**Comment by filmwork68** [comment]
Hey @josephdadams let me work on getting that to you, should have it shortly..

**Comment by filmwork68** [comment]
@josephdadams here is the descriptors as an HTML page. 
[aja-fs-HDR  FSHDR DIT CART  AJA FS Descriptors.zip](https://github.com/bitfocus/companion-module-requests/files/3811301/aja-fs-HDR.FSHDR.DIT.CART.AJA.FS.Descriptors.zip)

**Comment by scottyd5** [comment]
Also looking for an Aja FS2 module. Attached is a zipped HTML page. Thank you!

[FS2_ aja-fs2.zip](https://github.com/bitfocus/companion-module-requests/files/3855972/FS2_.aja-fs2.zip)

**Comment by josephdadams** [comment]
@scottyd5 I need you to send me the "descriptors" page. http://[ipaddress]/descriptors

Thanks!

**Comment by scottyd5** [comment]
> @scottyd5 I need you to send me the "descriptors" page. http://[ipaddress]/descriptors
> 
> Thanks!

Whoops. Here you go! 

[aja-fs2-Descriptors.zip](https://github.com/bitfocus/companion-module-requests/files/3870388/aja-fs2-Descriptors.zip)

**Comment by eicdw** [comment]
Hi @josephdadams - Any progress on a working module for FSHDR?
I also wonder if this would work for a FS-1X. I have a mix of the two.
Thank you

**Comment by josephdadams** [comment]
I looked at the protocol/commands but never wrote anything.

**Comment by gin-sir** [comment]
Hello all, as I can see, there're some people interested in this project, like me. I'd like to support you. I have the chance to have an Aja FS4 device at my disposal for testing. My development skills are limited, but we are interested in being able to control the aja colour correction devices, such as the FS2 or FS4.
If you are still interested in developing this topic, please contact me privately to set up a remote session (teamviewer or similar) to test the developments directly on the hardware.

**Comment by samuel2nd** [comment]
Hello, I also would be interested in having a module for this, I actually have this working with companion generic http module, doing setting like input selection, output format selection, procamp, and color correction adjustments. I have been doing this for awhile without a module, and have spreadsheets of commands, with and without companion variables.

**Comment by Bidz93** [comment]
Hi @samuel2nd - are you able to share any of this?

**Comment by QuandaryKD** [comment]
Is it possible to change selected LUT on the FSHDR using companion. Is this possible through the generic module or could it be written into the AJA FS module?

**Comment by samuel2nd** [comment]
Here is a link to a google sheet of the options I am using with the generic HTTP module. I have tested these on the AJA FS4, FSHDR , and FS2 units, some adjustments needed for FS2 unit. With my build it is a large build, 13 variables and corresponding triggers for for 1 input of a FS4 and I have 30inputs 7 FS4 units and 1 FS2, so 390 variables and triggers to have full procamp and color correction control. I do not currently have pages for all 30 built currently have pages for 1-12 fully functional , and have all variables and triggers in for all 30.
https://docs.google.com/spreadsheets/d/1ZT2A6kbJFlR0D7-rOjgAqyP3CCqnadORWLozJPg9t8s/edit

**Comment by samuel2nd** [comment]
https://github.com/user-attachments/assets/af9b5153-85cb-4947-a450-5c62cdd7e966

**Comment by samuel2nd** [comment]
https://github.com/user-attachments/assets/3673484f-7749-4549-813c-5de193c8e47f

**Comment by huntson** [comment]
> trim.FE96FFAA-94F9-419B-93FE-8786D520CC21.MOV

This is amazing.  Do you have a sample config you can share with the class to help those of us who are less-knowledgeable get off the ground?

**Comment by MichaelEdwards83** [comment]
Anyone manage to change inputs with companion?

**Comment by daniep01** [comment]
I now have occasional access to a FS4 and FS HDR so could work on this module if there is still a need for it

**Comment by ECLIPSE-EIC-coder** [comment]
@samuel2nd would love to connect have a few questions about implementing your method.

**Comment by samuel2nd** [comment]
> @samuel2nd would love to connect have a few questions about implementing your method.
> 

Absolutely, my email address is samuelstephens2nd@gmail.com, can find me on instagram and Facebook also!
